impeller/tessellator/tessellator.cc

```diff
@@ -218,6 +231,43 @@ Tessellator::Result Tessellator::Tessellate(
   return Result::kSuccess;
 }
 
+std::pair<std::vector<Point>, std::vector<uint16_t>>
+Tessellator::TessellateConvex(const Path& path, Scalar tolerance) {
+  std::vector<Point> output;
+  std::vector<uint16_t> indices;
+
+  point_buffer_->clear();
+  auto polyline =
+      path.CreatePolyline(tolerance, std::move(point_buffer_),
+                          [&](Path::Polyline::PointBufferPointer point_buffer) {
+                            point_buffer_ = std::move(point_buffer);
+                          });
+
+  for (auto j = 0u; j < polyline.contours.size(); j++) {
+    auto [start, end] = polyline.GetContourPointBounds(j);
+    auto center = (*polyline.points)[start];
+
+    // Some polygons will not self close and an additional triangle
+    // must be inserted, others will self close and we need to avoid
+    // inserting an extra triangle.
+    if ((*polyline.points)[end - 1] == (*polyline.points)[start]) {
```

Comment:
Consider adding a method to `Polyline` that returns a reference to the point at a given index

Reply:
Done